src/cli/agent.rs

```diff
@@ -48,6 +48,10 @@ pub struct Agent {
     /// The configuration source for a management server.
     #[clap(subcommand)]
     pub provider: Option<crate::config::Providers>,
+    /// If specified, will use the first node address found with the specified
+    /// type
+    #[clap(long)]
+    pub address_type: Option<String>,
```

Comment:
If we want to later add an extra argument to say specify ipv4 vs ipv6 selection if there are multiple addresses of the same type (rather than just "the first one")?

Basically wondering if this should be `address_key`, since if we want to be able to specify by address type, it will probably be `address_type` at that point?

Also - should we do both in this PR now, since we're here?

Reply:
Sure I can do that I guess. But that's slightly confusing given that the nodeaddress has address and type, so using address-key to target the type feels unintuitive. But maybe that doesn't matter due to the distance between the input and the implementation.

Reply:
I think how I would implement it is as a separate argument to the CLI (e.g. `--prefer-ip-version=<ipv6/ipv4>`) in addition to the address-type argument, which requires you to also specify the address-type argument to reduce the number of edge cases.